src/DataMatrix.java

```diff
@@ -1,121 +1,154 @@
       }
-    
+
    }
-   
-   public DataMatrix(String text)//text only
+
+   public DataMatrix(String text)// text only
    {
       actualWidth = 0;
       actualHeight = 0;
       image = new BarcodeImage();
-      
-      if(readText(text) == false)
+
+      if (readText(text) == false)
          text = "";
    }
-   
-   //Accessors - Bryce
-   public int getActualWidth()
-   {
+
+   // Accessors - Bryce
+   public int getActualWidth() {
       return actualWidth;
    }
-   
-   public int getActualHeight()
-   {
+
+   public int getActualHeight() {
       return actualHeight;
    }
-   
-   //Jeannie
-   public boolean scan(BarcodeImage bc)
-   {
+
+   // Jeannie
+   public boolean scan(BarcodeImage bc) {
       // TODO Auto-generated method stub
       return false;
    }
 
+   // passing a String parameter to readText method.
+   // returning false if text is null
+   public boolean readText(String text) {
+      if (text == null)
+         return false;
 
-   //Harsandeep
-   public boolean readText(String text)
-   {
-      // TODO Auto-generated method stub
-      return false;
+      this.text = text;
+      return true;
    }
 
-   //Jeannie
-   public boolean generateImageFromText()
-   {
+   // Jeannie
+   public boolean generateImageFromText() {
       // TODO Auto-generated method stub
       return false;
    }
 
-   //Harsandeep
-   public boolean translateImageToText()
-   {
-      // TODO Auto-generated method stub
-      return false;
+// reading char from each column in image
+   private char readCharFromCol(int col) {
+      int readChar = 0;
+      int fromCol = 0;
+
+      for (int i = BarcodeImage.MAX_HEIGHT - 2; i > BarcodeImage.MAX_HEIGHT - actualHeight; i--) {
+         if (this.image.getPixel(i, col)) {
+            readChar = readChar + (int) Math.pow(2, fromCol);
+         }
+         fromCol++;
+      }
+      return (char) (readChar);
    }
 
-   //Jeannie
-   public void displayTextToConsole()
-   {
+// translating image to text by decoding each column
+   public boolean translateImageToText() {
+      char colChar[] = new char[actualWidth - 1];
+      for (int i = 1; i < actualWidth - 1; i++) {
+         colChar[i] = readCharFromCol(i);
+      }
+      String colString = new String(colChar);
+      text = colString;
+      return true;
+   }
+
+   // Jeannie
+   public void displayTextToConsole() {
       // TODO Auto-generated method stub
 
    }
 
-   //Bryce
-   public void displayImageToConsole()
-   {
+   // Bryce
+   public void displayImageToConsole() {
       // TODO Auto-generated method stub
 
    }
-   
-   //Harsandeep
-   private void cleanImage()
-   {
-      
-   }
-   
-   //Bryce
-   private int computeSignalWidth()
-   {
-      
+
+   // move the signal to the lower-left of the larger 2D array 
+   //with cleanImage method.
+   private void cleanImage() {
+      int startRow = 0;
+      int startColumn = 0;
+		boolean spine = false;
+
+		for (int i = BarcodeImage.MAX_HEIGHT; i > 0; i--) {
```

Comment:
line 120: your for loop should be `for (int i = BarcodeImage.Max_Height-1` instead of Height.